Moved API helpers from AK test class to utils, resolved few issues

abalakh opened the pull request:
Few API helpers were implemented inside activation keys test class, moved them to api utils as i need them in different module. Also resolved few small issues with them like too small default timeout for rh repo sync etc.
Test results (forgot to install fake manifest at first):
```python
pytest -v tests/foreman/ui/test_activationkey.py
============================= test session starts ==============================
platform darwin -- Python 2.7.13, pytest-3.3.2, py-1.5.2, pluggy-0.6.0 -- /Users/andrii/workspace/env/bin/python
cachedir: .cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /Users/andrii/workspace/robottelo, inifile:
plugins: wait-for-1.0.9, xdist-1.21.0, services-1.2.1, mock-1.6.3, forked-0.2, cov-2.5.1
collected 41 items
2018-03-16 11:44:18 - conftest - DEBUG - BZ deselect is disabled in settings


tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_negative_copy <- robottelo/decorators/__init__.py PASSED [  2%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_negative_create_with_invalid_limit PASSED [  4%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_negative_create_with_invalid_name PASSED [  7%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_negative_delete PASSED [  9%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_negative_update_limit PASSED [ 12%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_negative_update_name PASSED [ 14%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_negative_usage_limit <- robottelo/decorators/__init__.py PASSED [ 17%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_access_non_admin_user <- robottelo/decorators/__init__.py PASSED [ 19%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_add_custom_product <- robottelo/decorators/__init__.py PASSED [ 21%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_add_host <- robottelo/decorators/__init__.py PASSED [ 24%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_add_host_collection_non_admin PASSED [ 26%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_add_multiple_aks_to_system <- robottelo/decorators/__init__.py PASSED [ 29%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_add_rh_and_custom_products <- robottelo/decorators/__init__.py FAILED [ 31%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_add_rh_product <- robottelo/decorators/__init__.py FAILED [ 34%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_copy <- robottelo/decorators/__init__.py PASSED [ 36%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_create_with_cv <- robottelo/decorators/__init__.py PASSED [ 39%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_create_with_description PASSED [ 41%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_create_with_envs <- robottelo/decorators/__init__.py PASSED [ 43%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_create_with_host_collection PASSED [ 46%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_create_with_name PASSED [ 48%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_create_with_usage_limit PASSED [ 51%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_delete PASSED [ 53%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_delete_manifest <- robottelo/decorators/__init__.py FAILED [ 56%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_delete_with_cv <- robottelo/decorators/__init__.py PASSED [ 58%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_delete_with_env <- robottelo/decorators/__init__.py PASSED [ 60%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_delete_with_system <- robottelo/decorators/__init__.py PASSED [ 63%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_end_to_end <- robottelo/decorators/__init__.py SKIPPED [ 65%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_fetch_product_content <- robottelo/decorators/__init__.py FAILED [ 68%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_host_associations <- robottelo/decorators/__init__.py PASSED [ 70%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_open_associated_host <- robottelo/decorators/__init__.py PASSED [ 73%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_remove_host_collection_non_admin PASSED [ 75%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_remove_user PASSED [ 78%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_search_scoped <- robottelo/decorators/__init__.py PASSED [ 80%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_service_level_subscription_with_custom_product <- robottelo/decorators/__init__.py FAILED [ 82%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_update_cv <- robottelo/decorators/__init__.py FAILED [ 85%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_update_description PASSED [ 87%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_update_env <- robottelo/decorators/__init__.py FAILED [ 90%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_update_limit PASSED [ 92%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_update_limit_to_unlimited PASSED [ 95%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_update_name PASSED [ 97%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_update_rh_product <- robottelo/decorators/__init__.py FAILED [100%]
============== 8 failed, 32 passed, 1 skipped in 3286.49 seconds ===============
```
Re-executed failed tests after installing fake cert:
```python
pytest -v tests/foreman/ui/test_activationkey.py -k "test_positive_add_rh_and_custom_products or test_positive_add_rh_product or test_positive_delete_manifest or test_positive_fetch_product_content or test_positive_service_level_subscription_with_custom_product or test_positive_update_cv or test_positive_update_env or test_positive_update_rh_product"
============================= test session starts ==============================
platform darwin -- Python 2.7.13, pytest-3.3.2, py-1.5.2, pluggy-0.6.0 -- /Users/andrii/workspace/env/bin/python
cachedir: .cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /Users/andrii/workspace/robottelo, inifile:
plugins: wait-for-1.0.9, xdist-1.21.0, services-1.2.1, mock-1.6.3, forked-0.2, cov-2.5.1
collected 41 items
2018-03-16 12:54:09 - conftest - DEBUG - BZ deselect is disabled in settings


tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_add_rh_and_custom_products <- robottelo/decorators/__init__.py PASSED [ 12%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_add_rh_product <- robottelo/decorators/__init__.py PASSED [ 25%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_delete_manifest <- robottelo/decorators/__init__.py PASSED [ 37%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_fetch_product_content <- robottelo/decorators/__init__.py PASSED [ 50%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_service_level_subscription_with_custom_product <- robottelo/decorators/__init__.py PASSED [ 62%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_update_cv <- robottelo/decorators/__init__.py PASSED [ 75%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_update_env <- robottelo/decorators/__init__.py PASSED [ 87%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_update_rh_product <- robottelo/decorators/__init__.py PASSED [100%]

============================= 33 tests deselected ==============================
================== 8 passed, 33 deselected in 1464.65 seconds ==================
```

oshtaier commented on robottelo/api/utils.py:
```diff
@@ -192,6 +194,71 @@ def delete_puppet_class(puppetclass_name, puppet_module=None,
+    return repo_id
+
+
+def cv_publish_promote(name, env_name, repo_id, org_id):
```
repository definitely is not mandatory for publish/promote procedure, so it is necessary to have options. That looks like required information:
```
    org = entities.Organization().create()
    lce = entities.LifecycleEnvironment(organization=org).create()
    content_view = entities.ContentView(organization=org).create()
    content_view.publish()
    promote(content_view.read().version[0], lce.id)
```
All other things are optional.
Please think about that as it is right time for that

abalakh commented:
@oshtaier done.
```python
pytest -v tests/foreman/ui/test_activationkey.py -k test_positive_add_custom_product
============================= test session starts ==============================
platform darwin -- Python 2.7.13, pytest-3.3.2, py-1.5.2, pluggy-0.6.0 -- /Users/andrii/workspace/env/bin/python
cachedir: .cache
addressed comments
shared_function enabled - OFF - scope:  - storage: file
rootdir: /Users/andrii/workspace/robottelo, inifile:
plugins: wait-for-1.0.9, xdist-1.21.0, services-1.2.1, mock-1.6.3, forked-0.2, cov-2.5.1
collected 41 items
2018-03-16 13:56:07 - conftest - DEBUG - BZ deselect is disabled in settings


tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_add_custom_product <- robottelo/decorators/__init__.py PASSED [100%]

============================= 40 tests deselected ==============================
================== 1 passed, 40 deselected in 127.49 seconds ===================
```

abalakh commented:
```python
pytest -v tests/foreman/ui/test_activationkey.py -k test_positive_add_custom_product
============================= test session starts ==============================
platform darwin -- Python 2.7.13, pytest-3.3.2, py-1.5.2, pluggy-0.6.0 -- /Users/andrii/workspace/env/bin/python
cachedir: .cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /Users/andrii/workspace/robottelo, inifile:
plugins: wait-for-1.0.9, xdist-1.21.0, services-1.2.1, mock-1.6.3, forked-0.2, cov-2.5.1
collected 41 items
2018-03-16 15:24:21 - conftest - DEBUG - BZ deselect is disabled in settings


tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_add_custom_product <- robottelo/decorators/__init__.py PASSED [100%]

============================= 40 tests deselected ==============================
================== 1 passed, 40 deselected in 114.24 seconds ===================
```

abalakh commented:
```python
pytest -v tests/foreman/ui/test_activationkey.py
============================= test session starts ==============================
platform darwin -- Python 2.7.13, pytest-3.3.2, py-1.5.2, pluggy-0.6.0 -- /Users/andrii/workspace/env/bin/python
cachedir: .cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /Users/andrii/workspace/robottelo, inifile:
plugins: wait-for-1.0.9, xdist-1.21.0, services-1.2.1, mock-1.6.3, forked-0.2, cov-2.5.1
collected 41 items
2018-03-16 16:07:16 - conftest - DEBUG - BZ deselect is disabled in settings


tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_negative_copy <- robottelo/decorators/__init__.py PASSED [  2%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_negative_create_with_invalid_limit PASSED [  4%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_negative_create_with_invalid_name PASSED [  7%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_negative_delete PASSED [  9%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_negative_update_limit PASSED [ 12%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_negative_update_name PASSED [ 14%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_negative_usage_limit <- robottelo/decorators/__init__.py PASSED [ 17%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_access_non_admin_user <- robottelo/decorators/__init__.py PASSED [ 19%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_add_custom_product <- robottelo/decorators/__init__.py PASSED [ 21%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_add_host <- robottelo/decorators/__init__.py PASSED [ 24%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_add_host_collection_non_admin PASSED [ 26%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_add_multiple_aks_to_system <- robottelo/decorators/__init__.py PASSED [ 29%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_add_rh_and_custom_products <- robottelo/decorators/__init__.py PASSED [ 31%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_add_rh_product <- robottelo/decorators/__init__.py PASSED [ 34%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_copy <- robottelo/decorators/__init__.py PASSED [ 36%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_create_with_cv <- robottelo/decorators/__init__.py PASSED [ 39%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_create_with_description PASSED [ 41%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_create_with_envs <- robottelo/decorators/__init__.py PASSED [ 43%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_create_with_host_collection PASSED [ 46%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_create_with_name PASSED [ 48%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_create_with_usage_limit PASSED [ 51%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_delete PASSED [ 53%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_delete_manifest <- robottelo/decorators/__init__.py PASSED [ 56%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_delete_with_cv <- robottelo/decorators/__init__.py PASSED [ 58%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_delete_with_env <- robottelo/decorators/__init__.py PASSED [ 60%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_delete_with_system <- robottelo/decorators/__init__.py PASSED [ 63%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_end_to_end <- robottelo/decorators/__init__.py SKIPPED [ 65%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_fetch_product_content <- robottelo/decorators/__init__.py PASSED [ 68%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_host_associations <- robottelo/decorators/__init__.py PASSED [ 70%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_open_associated_host <- robottelo/decorators/__init__.py PASSED [ 73%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_remove_host_collection_non_admin PASSED [ 75%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_remove_user PASSED [ 78%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_search_scoped <- robottelo/decorators/__init__.py PASSED [ 80%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_service_level_subscription_with_custom_product <- robottelo/decorators/__init__.py PASSED [ 82%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_update_cv <- robottelo/decorators/__init__.py FAILED [ 85%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_update_description PASSED [ 87%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_update_env <- robottelo/decorators/__init__.py FAILED [ 90%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_update_limit PASSED [ 92%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_update_limit_to_unlimited PASSED [ 95%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_update_name PASSED [ 97%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_update_rh_product <- robottelo/decorators/__init__.py PASSED [100%]
============== 2 failed, 38 passed, 1 skipped in 4746.06 seconds ===============
pytest -v tests/foreman/ui/test_activationkey.py -k "test_positive_update_cv or test_positive_update_env"
============================= test session starts ==============================
platform darwin -- Python 2.7.13, pytest-3.3.2, py-1.5.2, pluggy-0.6.0 -- /Users/andrii/workspace/env/bin/python
cachedir: .cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /Users/andrii/workspace/robottelo, inifile:
plugins: wait-for-1.0.9, xdist-1.21.0, services-1.2.1, mock-1.6.3, forked-0.2, cov-2.5.1
collected 41 items
2018-03-16 17:31:54 - conftest - DEBUG - BZ deselect is disabled in settings


tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_update_cv <- robottelo/decorators/__init__.py PASSED [ 50%]
tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_update_env <- robottelo/decorators/__init__.py PASSED [100%]

============================= 39 tests deselected ==============================
================== 2 passed, 39 deselected in 346.70 seconds ===================
```
